prov/efa/src/rxr/rxr_ep.c

```diff
@@ -1618,8 +1639,10 @@ int rxr_endpoint(struct fid_domain *domain, struct fi_info *info,
 
 	efa_domain = container_of(rxr_domain->rdm_domain, struct efa_domain,
 				  util_domain.domain_fid);
-	/* Open shm provider's endpoint */
-	if (rxr_env.enable_shm_transfer) {
+
+	rxr_ep->use_shm = rxr_ep_use_shm(info);
```

Comment:
If I understand correctly, we are assuming the app could pass different fi_info to different EPs, and expect them to behave differently on shm usage. Correct?

Reply:
Yes.